overlord/auth/auth.go

```diff
@@ -84,6 +86,51 @@ func User(st *state.State, id int) (*UserState, error) {
 	return nil, fmt.Errorf("invalid user")
 }
 
+// GetUserFromRequest extracts user information from request and return the respective user in state, if valid
+func GetUserFromRequest(st *state.State, req *http.Request) (*UserState, error) {
```

Comment:
`s/Get//`, and let's please move this into the daemon. This is about http, so the overlord isn't a good home.